src/util/string.cpp

```diff
@@ -8,6 +8,25 @@
 #include <string>
 
 namespace util {
+
+// Compile-time sanity checks
+static_assert([] {
+    ConstevalFormatString<0>::CheckNumFormatSpecifiers("");
+    ConstevalFormatString<0>::CheckNumFormatSpecifiers("%");
+    ConstevalFormatString<0>::CheckNumFormatSpecifiers("%%");
+    ConstevalFormatString<0>::CheckNumFormatSpecifiers("%%%");
```

Comment:
shouldn't we warn in this case as well, i.e. an unfinished %?
The tinyformat tests use two trailing % when it's an actual %:
https://github.com/c42f/tinyformat/blob/master/tinyformat_test.cpp#L144